src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/FSharp/FSharpProjectSelector.cs

```diff
@@ -1,19 +1,47 @@
-﻿using System;
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.ComponentModel.Composition;
 using System.Runtime.InteropServices;
 using System.Xml;
 using System.Xml.Linq;
 using System.Xml.XPath;
 using Microsoft.VisualStudio.Shell;
 using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Threading;
+using IAsyncServiceProvider = Microsoft.VisualStudio.Shell.IAsyncServiceProvider;
+using Task = System.Threading.Tasks.Task;
 
 namespace Microsoft.VisualStudio.ProjectSystem.VS.FSharp
 {
+    [Export(typeof(IPackageService))]
     [Guid("E720DAD0-1854-47FC-93AF-E719B54B84E6")]
     [ProvideObject(typeof(FSharpProjectSelector), RegisterUsing = RegistrationMethod.CodeBase)]
-    internal sealed class FSharpProjectSelector : IVsProjectSelector
+    internal sealed class FSharpProjectSelector : IVsProjectSelector, IPackageService, IDisposable
     {
         private const string MSBuildXmlNamespace = "http://schemas.microsoft.com/developer/msbuild/2003";
 
+        private readonly JoinableTaskContext _context;
+        private IVsRegisterProjectSelector? _projectSelector;
+        private uint _cookie = VSConstants.VSCOOKIE_NIL;
+
+        public FSharpProjectSelector(JoinableTaskContext context)
```

Comment:
[ImportingCosntructor]

Reply:
Shouldn't an analyzer flag this?

Reply:
There is an analyzer for this, but we've turned it off.

https://github.com/dotnet/project-system/blob/master/.editorconfig#L332

Turning it on flags a bunch of diagnostics. I'll take a look at this in a follow up. Some of the diagnostics look wrong. If so I'll get some bugs filed.

Reply:
https://github.com/dotnet/roslyn-analyzers/issues/3025